File: Easy/459. Repeated Substring Pattern/459. Repeated Substring Pattern.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        n = len(s)

        for length in range(1, n // 2 + 1):
            if n % length == 0:
                repeat_count = n // length
                pattern = s[:length]
                constructed = pattern * repeat_count
                if constructed == s:
                    return True

        return False

File: Easy/459. Repeated Substring Pattern/459. Repeated Substring Pattern.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        int n = s.length();
        
        for (int len = 1; len <= n / 2; len++) {
            if (n % len == 0) {
                int repeatCount = n / len;
                String pattern = s.substring(0, len);
                StringBuilder constructed = new StringBuilder();
                for (int i = 0; i < repeatCount; i++) {
                    constructed.append(pattern);
                }
                if (constructed.toString().equals(s)) {
                    return true;
                }
            }
        }
        
        return false;
    }
}
